config/routes.rb

```diff
@@ -1,6 +1,13 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  root to: 'home#index'
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show, :create, :destroy] do
+      resources :comments, only: [:create]
+      
+      # Like a post
+      post 'like', to: 'likes#create', as: 'like'
+      delete 'unlike', to: 'likes#destroy', as: 'unlike'
+    end
+  end
 end
```

Comment:
- [ ] Kindly ensure this route `GET   /posts/new `  renders a form to create a `new` post that can be saved into the database.
- [ ] This route `GET /posts/:post_id/comments/new` should also  render a form to create a comment